File: src/main/kotlin/dev/jefy/connectpro/marketplace/applicaion/command/JobApplicationCommand.java
package dev.jefy.connectpro.marketplace.applicaion.command;

import org.jspecify.annotations.NullMarked;

import dev.jefy.connectpro.marketplace.applicaion.dtos.JobApplicationRequest;
import dev.jefy.connectpro.marketplace.domain.vo.JobApplicationId;

/**
 * @author Jôph Yamba
 */

@NullMarked
public interface JobApplicationCommand {
    
    JobApplicationId apply(JobApplicationRequest request);
    JobApplicationId accept(JobApplicationId applicationId);
    JobApplicationId reject(JobApplicationId applicationId);
}

File: src/main/kotlin/dev/jefy/connectpro/marketplace/application/command/JobApplicationCommand.kt
package dev.jefy.connectpro.marketplace.application.command

import dev.jefy.connectpro.marketplace.application.dtos.JobApplicationRequest
import dev.jefy.connectpro.marketplace.domain.vo.JobApplicationId
import org.jspecify.annotations.NullMarked

/**
 * @author Jôph Yamba
 */

interface JobApplicationCommand {
    fun apply(request: JobApplicationRequest): JobApplicationId
    fun accept(applicationId: JobApplicationId): JobApplicationId
    fun reject(applicationId: JobApplicationId): JobApplicationId
}
